### DISCUSSION TITLE
Compatibility with different modes

### DISCUSSION THREAD

**Discussion by NullExir** [Joshua-Ashton/VPhysics-Jolt#206] [Q&A]
Hi, i just downloaded the binaries from the release and extracted vphysics.dll from `vphysics_jolt_0.11_sdk2013-mp_win32.zip` into the Dedicated server of l4d2 `Left 4 Dead 2/bin`, Running with the following command `.\srcds.exe -console -game left4dead2 -dev -insecure`

OS: windows 10
version: 22H2
srcds version:
```stdout
Version 2.2.3.7 (left4dead2)
Network Version 2.1.0.0
Exe build: 08:21:17 Feb 20 2024 (9214) (550)
```

There isn't any output or crash dumps

I'm asking mainly after watching this segment of [this video](https://youtu.be/OIvbxikwEbU?t=146) (replacing tf2's default physics engine with the Jolt engine), so it should be possible

The only issue that is relevant is this one #168

**Comment by Sectonid** [comment]
> I'm asking mainly after watching this segment of [this video](https://youtu.be/OIvbxikwEbU?t=146)

There wasn't any L4D/L4D2 footage on the video, plus L4D2 uses separate branch of Source Engine than Source 2013. This basically means that the engine stuff was changed to the point it's not compatible with previous branch, so you'll have to port VPhysics Jolt to L4D branch.

<img width="1404" height="103" alt="image" src="https://github.com/user-attachments/assets/60bccee0-6395-4c53-8548-93a1c4628515" />

But Valve Software never open-sourced any of their engines code (GoldSRC, Source and all of its branches, Source 2), so you'll have to port it yourself if you:
* Got license for using Source Engine's closed source code and signed NDA for that
* Know C++ coding
* Are less vulnerable to hacker/malware/trojan/etc. attacks (Valve wouldn't like to see another engine branch leak)

Unfortunately, there's no chance that you can get license for Source Engine nowadays, so you can only hope that someone with a license will port VPhysics Jolt to L4D branch.
